kestrel-opentelemetry-metrics/pom.xml

```
@@ -31,15 +37,33 @@
             <artifactId>kestrel-core</artifactId>
             <version>${project.version}</version>
         </dependency>
-
         <dependency>
-            <groupId>io.prometheus</groupId>
-            <artifactId>simpleclient</artifactId>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.opentelemetry</groupId>
+            <artifactId>opentelemetry-extension-kotlin</artifactId>
         </dependency>
         <dependency>
             <groupId>org.jetbrains.kotlin</groupId>
             <artifactId>kotlin-stdlib</artifactId>
         </dependency>
+        <dependency>
+            <scope>test</scope>
+            <groupId>io.kotest</groupId>
+            <artifactId>kotest-framework-api-jvm</artifactId>
+        </dependency>
+        <dependency>
+            <scope>test</scope>
+            <groupId>io.kotest</groupId>
+            <artifactId>kotest-runner-junit5-jvm</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>kestrel-http-event-streams</artifactId>
+            <version>${project.version}</version>
+        </dependency>
```

Can be simplied and also not use transitive dependencies.

```suggestion
            <groupId>${project.groupId}</groupId>
            <artifactId>kestrel-http-event-streams</artifactId>
            <version>${project.version}</version>
        </dependency>

        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-context</artifactId>
        </dependency>
        <dependency>
            <groupId>org.jetbrains.kotlin</groupId>
            <artifactId>kotlin-stdlib</artifactId>
        </dependency>
        <dependency>
            <groupId>org.slf4j</groupId>
            <artifactId>slf4j-api</artifactId>
        </dependency>
```

Btw, is opentelemetry-extension-kotlin needed by the consumers of this library?

---

This was a part of attempting to get tracing working, which I've decided to drop, so I'll remove. 

The extension adds a coroutine context key (think a threadlocal for coroutines), this will allow us to manage trace scopes over multiple coroutine jobs

---

Cleared up I believe